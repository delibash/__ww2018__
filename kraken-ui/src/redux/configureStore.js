import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
import { createLogger } from 'redux-logger';
import * as reducers from './ducks';
import rootEpic from './epics';

const epicMiddleware = createEpicMiddleware();
const rootReducer = combineReducers(reducers);

const middleware = [
  epicMiddleware
];

const logger = createLogger({
  level: 'info',
  duration: true,
  collapsed: (getState, action, logs) => !logs.error,
  diff: true
});

export default function configureStore(initialState) {

  if (process.env.NODE_ENV === 'development') {
    middleware.push(logger);
  }

  const composedStore = compose(
    applyMiddleware(
      ...middleware,
    )(createStore)(
      rootReducer,
      initialState,
      window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    )
  );

  epicMiddleware.run(rootEpic);

  return composedStore;
}