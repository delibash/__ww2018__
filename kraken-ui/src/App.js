import React, { Component } from 'react';
import { Provider } from 'react-redux';
import configureStore from './redux/configureStore';
import Routes from './routes';

import {
  ResetStyles,
} from './global/styles';

const store = configureStore();

class App extends Component {
  render () {
    return (
      <>
        <ResetStyles />
        <Provider store={store}>
          <Routes />
        </Provider>
      </>
    );
  }
}

App.propTypes = {};

export default App;
