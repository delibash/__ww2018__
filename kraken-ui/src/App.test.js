import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import App from './App';
import { Provider } from 'react-redux';
import configureStore from './redux/configureStore';
import {
  ResetStyles,
} from './global/styles';

const store = configureStore();

test('Renders AddNodes', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <ResetStyles />
      <App />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});