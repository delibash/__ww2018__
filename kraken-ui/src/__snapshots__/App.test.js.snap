// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders AddNodes 1`] = `
.c4 {
  margin: 0 -1.8rem;
}

.c4 .react-tabs__tab-panel {
  padding: 1.8rem;
  min-height: calc(100vh - 76px);
  margin-top: -.1rem;
  box-shadow: inset 0 0 0 .1rem rgba(0,0,0,.25);
}

.c4 .react-tabs__tab-list {
  border-bottom: .1rem solid rgba(0,0,0,.25);
  margin-bottom: 0;
}

.c4 .react-tabs__tab {
  border: 0;
  margin-left: 1.3rem;
  box-shadow: inset 0 0 0 .1rem rgba(0,0,0,.25);
}

.c4 .react-tabs__tab--selected {
  border-radius: .2rem .2rem 0 0;
  background-color: rgba(74,74,74,1);
  color: rgba(255,255,255,1);
}

.c2 {
  margin: 1rem 1.8rem 1.8rem;
}

.c3 {
  margin: 0 1.8rem;
}

.c9 {
  position: relative;
}

.c10 {
  min-height: 3.8rem;
  padding: .2rem .8rem;
  border: 0;
  border-radius: .3rem;
  box-shadow: inset 0 0 0 .1rem rgba(0,0,0,.25);
  font-size: 1.3rem;
  width: 100%;
}

.c11 {
  position: absolute;
  right: .1rem;
  top: .1rem;
  background: rgba(234,236,242,1);
  color: rgba(255,255,255,1);
  border: 0 none;
  min-height: 3.6rem;
  border-radius: 0 .3rem .3rem 0;
  cursor: pointer;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1.8rem;
}

.c5 strong ~ div {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 strong ~ div:not(:last-child) {
  margin: 0 1.8rem;
}

.c5 strong ~ div:last-child {
  margin-left: 1.8rem;
}

.c0 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: repeat(2,1fr);
  grid-gap: 0 0;
  background-color: transparent;
  min-height: 100vh;
}

.c7 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: 1fr;
  grid-gap: 0;
  background-color: transparent;
  min-height: initial;
  position: relative;
}

.c1 {
  grid-column: auto;
  grid-row: auto;
}

.c6 {
  grid-column: auto;
  grid-row: auto;
  background-color: rgba(234,236,242,1);
}

.c8 {
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: 2fr 1fr;
}

@media (max-width:36em) {
  .c0 {
    grid-template-columns: auto;
    grid-template-rows: repeat(2,1fr);
  }
}

@media (max-width:36em) {
  .c7 {
    grid-template-columns: auto;
    grid-template-rows: repeat(2,1fr);
  }
}

<section
  className="c0"
>
  <section
    className="c1"
  >
    <h2
      className="c2"
    >
      Add to Knowledge Graph
    </h2>
    <section
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="react-tabs"
          data-tabs={true}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <ul
            className="react-tabs__tab-list"
            role="tablist"
          >
            <li
              aria-controls="react-tabs-1"
              aria-disabled="false"
              aria-selected="true"
              className="react-tabs__tab react-tabs__tab--selected"
              id="react-tabs-0"
              role="tab"
              tabIndex="0"
            >
              Add Node(s)
            </li>
            <li
              aria-controls="react-tabs-3"
              aria-disabled="false"
              aria-selected="false"
              className="react-tabs__tab"
              id="react-tabs-2"
              role="tab"
              tabIndex={null}
            >
              Upload Nodes(Bulk)
            </li>
          </ul>
          <div
            aria-labelledby="react-tabs-0"
            className="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-1"
            role="tabpanel"
          >
            <section>
              <label
                className="c5"
              >
                <strong>
                  Add:
                </strong>
                <div
                  className="css-10nd86i"
                  onKeyDown={[Function]}
                >
                  <div
                    className="css-cgevzf"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <div
                      className="css-1hwfws3"
                    >
                      <div
                        className="css-1492t68"
                      >
                        Select One
                      </div>
                      <div
                        className="css-1g6gooi"
                      >
                        <div
                          className=""
                          style={
                            Object {
                              "display": "inline-block",
                            }
                          }
                        >
                          <input
                            aria-autocomplete="list"
                            autoCapitalize="none"
                            autoComplete="off"
                            autoCorrect="off"
                            disabled={false}
                            id="react-select-2-input"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            spellCheck="false"
                            style={
                              Object {
                                "background": 0,
                                "border": 0,
                                "boxSizing": "content-box",
                                "color": "inherit",
                                "fontSize": "inherit",
                                "opacity": 1,
                                "outline": 0,
                                "padding": 0,
                                "width": "1px",
                              }
                            }
                            tabIndex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style={
                              Object {
                                "height": 0,
                                "left": 0,
                                "overflow": "scroll",
                                "position": "absolute",
                                "top": 0,
                                "visibility": "hidden",
                                "whiteSpace": "pre",
                              }
                            }
                          >
                            
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      className="css-1wy0on6"
                    >
                      <span
                        className="css-d8oujb"
                      />
                      <div
                        aria-hidden="true"
                        className="css-1ep9fjw"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <svg
                          aria-hidden="true"
                          className="css-19bqh2r"
                          focusable="false"
                          height={20}
                          viewBox="0 0 20 20"
                          width={20}
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </label>
            </section>
          </div>
          <div
            aria-labelledby="react-tabs-2"
            className="react-tabs__tab-panel"
            id="react-tabs-3"
            role="tabpanel"
          />
        </div>
      </div>
    </section>
  </section>
  <section
    className="c6"
  >
    <h2
      className="c2"
    >
      Search Knowledge Graph
    </h2>
    <section
      className="c3"
    >
      <section
        className="c7"
      >
        <form
          className="c8"
          onSubmit={[Function]}
        >
          <label
            className="c9"
            htmlFor="search-KG"
          >
            <input
              className="c10"
              id="search-KG"
              onChange={[Function]}
              placeholder="Search KG"
            />
            <button
              className="c11"
              disabled={true}
              type="submit"
            >
              <i
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "
  <svg
    width=\\"20\\"
    height=\\"20\\"
    viewBox=\\"0 0 20 20\\"
    fill=\\"white\\"
  >
    <path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"/>
    <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"/>
  </svg>
  ",
                  }
                }
                src={null}
              />
            </button>
          </label>
          <div
            className="css-10nd86i"
            onKeyDown={[Function]}
          >
            <div
              className="css-cgevzf"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className="css-1hwfws3"
              >
                <div
                  className="css-1492t68"
                >
                  Type of Node
                </div>
                <div
                  className="css-1g6gooi"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-3-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="css-1wy0on6"
              >
                <span
                  className="css-d8oujb"
                />
                <div
                  aria-hidden="true"
                  className="css-1ep9fjw"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-19bqh2r"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </form>
      </section>
    </section>
  </section>
</section>
`;
